client-runtime/src/main/java/com/microsoft/rest/v2/http/RxNettyClient.java

```diff
@@ -64,10 +73,21 @@ private SSLEngine getSSLEngine(String host) {
             rxnHeaders.put("Content-Length", Collections.<Object>singleton(String.valueOf(body.length())));
         }
 
+        // Is caching of rxnetty's HttpClient preferred for performance?
         boolean isSecure = "https".equalsIgnoreCase(uri.getScheme());
         io.reactivex.netty.protocol.http.client.HttpClient<ByteBuf, ByteBuf> rxnClient =
                 io.reactivex.netty.protocol.http.client.HttpClient.newClient(uri.getHost(), isSecure ? 443 : 80);
 
+        for (int i = 0; i < handlerConfigs.length; i++) {
+            ChannelHandlerConfig config = handlerConfigs[i];
+            if (config.mayBlock()) {
+                EventExecutorGroup executorGroup = new DefaultEventExecutorGroup(1);
+                rxnClient = rxnClient.addChannelHandlerLast(executorGroup, "az-client-handler-" + i, config.factory());
+            } else {
+                rxnClient = rxnClient.addChannelHandlerLast("az-client-handler-" + i, config.factory());
```

Comment:
Calculate "az-client-handler-" + I and config.factory() before the if statement. That way they'll be easier to change for both branches of the if statement.